### ISSUE TITLE
high load and no one is writing anymore [18a7b73e0882]

### ISSUE THREAD

**Issue by colttt** [koverstreet/bcachefs#427]
I'm running it on Debian 11, on a Dell PowerEdge R730.
I run:
```
fio --rw=randrw --name=test --numjobs=8 --runtime=36000 --iodepth=64 --size=120G --filename=/mnt/blubbs --group_reporting --time_based
```
after that bcachefs is still running with high CPU usage, I also wait 1hour after fio is finished..
```
top - 13:10:44 up  4:02,  3 users,  load average: 7.00, 7.06, 9.63
Tasks: 373 total,   3 running, 370 sleeping,   0 stopped,   0 zombie
%Cpu(s):  0.0 us,  6.4 sy,  0.0 ni, 93.6 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
MiB Mem :  32058.4 total,   6906.9 free,   2692.6 used,  22458.9 buff/cache
MiB Swap:   9023.0 total,   9022.0 free,      1.0 used.  28894.5 avail Mem 

    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND                                                                                                                                                                                                                                              
   1796 root      20   0       0      0      0 R  89.4   0.0   8:19.61 kworker/22:2+bcachefs                                                                                                                                                                                                                                
   2000 root      20   0       0      0      0 R  71.2   0.0   8:56.67 kworker/18:2+bcachefs                                                                                                                                                                                                                                
   1862 root      20   0       0      0      0 D  32.1   0.0   6:57.97 kworker/20:0+bcachefs                                                                                                                                                                                                                                
   2104 root      20   0       0      0      0 D  14.2   0.0   6:48.22 kworker/26:0+bcachefs                                                                                                                                                                                                                                
    259 root      20   0       0      0      0 D   3.6   0.0   5:05.22 kworker/28:1+bcachefs                                                                                                                                                                                                                                
   1859 root      20   0       0      0      0 D   3.6   0.0   7:02.21 kworker/24:2+bcachefs 
   
Overhead  Shared Object                Symbol                                                                                                                                                                                                                                                                               
  40.46%  [kernel]                     [k] osq_lock                                                                                                                                                                                                                                                                         
  23.69%  [kernel]                     [k] six_optimistic_spin                                                                                                                                                                                                                                                              
   2.61%  [kernel]                     [k] _find_next_bit                                                                                                                                                                                                                                                                   
   1.78%  [kernel]                     [k] bch2_btree_path_verify_locks                                                                                                                                                                                                                                                     
   1.36%  [kernel]                     [k] cpumask_next                                                                                                                                                                                                                                                                     
   0.95%  [kernel]                     [k] __bch2_trans_commit                                                                                                                                                                                                                                                              
   0.87%  [kernel]                     [k] bch2_btree_path_set_pos                                                                                                                                                                                                                                                          
   0.87%  [kernel]                     [k] six_trylock_write                                                                                                                                                                                                                                                                
   0.85%  [kernel]                     [k] __bch2_btree_node_alloc                                                                                                                                                                                                                                                          
   0.84%  [kernel]                     [k] bch2_btree_node_iter_init                                                                                                                                                                                                                                                        
   0.77%  [kernel]                     [k] bch2_trans_update_by_path_trace                                                                                                                                                                                                                                                  
   0.72%  [kernel]                     [k] bch2_btree_path_verify                                                                                                                                                                                                                                                           
   0.69%  [kernel]                     [k] bch2_path_get                                                                                                                                                                                                                                                                    
   0.68%  [kernel]                     [k] bch2_path_put                                                                                                                                                                                                                                                                    
   0.66%  [kernel]                     [k] six_trylock_intent                                                                                                                                                                                                                                                               
   0.61%  [kernel]                     [k] bch2_btree_path_verify_level                                                                                                                                                                                                                                                     
   0.60%  [kernel]                     [k] bch2_varint_decode_fast                                                                                                                                                                                                                                                          
   0.59%  [kernel]                     [k] six_unlock_write                                                                                                                                                                                                                                                                 
   0.58%  [kernel]                     [k] mutex_unlock                                                                                                                                                                                                                                                                     
   0.58%  [kernel]                     [k] bch2_btree_node_mem_alloc                                                                                                                                                                                                                                                        
   0.54%  [kernel]                     [k] btree_path_alloc               
   ```

**Version**
```
uname -a
Linux bcachefs 5.18.0-18a7b73e0882-bcachefs #1 SMP PREEMPT_DYNAMIC Thu Jun 23 08:47:13 CEST 2022 x86_64 GNU/Linux

bcachefs version
bcachefs tool version v0.1-542-gb4afc1f
```


**Generic info**
Provide the output of:
```
bcachefs format --verbose /dev/sdb /dev/sdc /dev/sdd /dev/sde /dev/sdf /dev/sdg /dev/sdh /dev/sdi /dev/sdj

bcachefs fs usage /mnt/
Filesystem: 20c5609b-403d-4e73-b4b9-efbda35fa7af
Size:                  4969049767424
Used:                   167935475712
Online reserved:                   0

Data type       Required/total  Devices
btree:          1/1             [sdh]                       42729472
btree:          1/1             [sdd]                       44040192
btree:          1/1             [sdb]                       46661632
btree:          1/1             [sdf]                       44302336
btree:          1/1             [sdj]                       46137344
btree:          1/1             [sdc]                       41680896
btree:          1/1             [sde]                       44564480
btree:          1/1             [sdg]                       44564480
btree:          1/1             [sdi]                       44040192
user:           1/1             [sdg]                    14319616000
user:           1/1             [sdc]                    14317887488
user:           1/1             [sdi]                    14315700224
user:           1/1             [sde]                    14316920832
user:           1/1             [sdh]                    14314274816
user:           1/1             [sdj]                    14316105728
user:           1/1             [sdb]                    14315065344
user:           1/1             [sdd]                    14317805568
user:           1/1             [sdf]                    14315642880

(no label) (device 0):           sdb              rw
                                data         buckets      fragmented
  free:                            0         1108556
  sb:                        3149824               7          520192
  journal:                4294967296            8192
  btree:                    46661632             101         6291456
  user:                  14315065344           27795       257519616
  cached:                          0               0
  parity:                          0               0
  stripe:                          0               0
  need_gc_gens:                    0               0
  need_discard:                    0               0
  erasure coded:                   0               0
  capacity:             600126783488         1144651

(no label) (device 1):           sdc              rw
                                data         buckets      fragmented
  free:                            0         1108558
  sb:                        3149824               7          520192
  journal:                4294967296            8192
  btree:                    41680896              94         7602176
  user:                  14317887488           27800       257318912
  cached:                          0               0
  parity:                          0               0
  stripe:                          0               0
  need_gc_gens:                    0               0
  need_discard:                    0               0
  erasure coded:                   0               0
  capacity:             600126783488         1144651

(no label) (device 2):           sdd              rw
                                data         buckets      fragmented
  free:                            0         1108552
  sb:                        3149824               7          520192
  journal:                4294967296            8192
  btree:                    44040192             100         8388608
  user:                  14317805568           27800       257400832
  cached:                          0               0
  parity:                          0               0
  stripe:                          0               0
  need_gc_gens:                    0               0
  need_discard:                    0               0
  erasure coded:                   0               0
  capacity:             600126783488         1144651

(no label) (device 3):           sde              rw
                                data         buckets      fragmented
  free:                            0         1108558
  sb:                        3149824               7          520192
  journal:                4294967296            8192
  btree:                    44564480              97         6291456
  user:                  14316920832           27797       256712704
  cached:                          0               0
  parity:                          0               0
  stripe:                          0               0
  need_gc_gens:                    0               0
  need_discard:                    0               0
  erasure coded:                   0               0
  capacity:             600126783488         1144651

(no label) (device 4):           sdf              rw
                                data         buckets      fragmented
  free:                            0         1108556
  sb:                        3149824               7          520192
  journal:                4294967296            8192
  btree:                    44302336             101         8650752
  user:                  14315642880           27795       256942080
  cached:                          0               0
  parity:                          0               0
  stripe:                          0               0
  need_gc_gens:                    0               0
  need_discard:                    0               0
  erasure coded:                   0               0
  capacity:             600126783488         1144651

(no label) (device 5):           sdg              rw
                                data         buckets      fragmented
  free:                            0         1108553
  sb:                        3149824               7          520192
  journal:                4294967296            8192
  btree:                    44564480              97         6291456
  user:                  14319616000           27802       256638976
  cached:                          0               0
  parity:                          0               0
  stripe:                          0               0
  need_gc_gens:                    0               0
  need_discard:                    0               0
  erasure coded:                   0               0
  capacity:             600126783488         1144651

(no label) (device 6):           sdh              rw
                                data         buckets      fragmented
  free:                            0         1108563
  sb:                        3149824               7          520192
  journal:                4294967296            8192
  btree:                    42729472              91         4980736
  user:                  14314274816           27798       259883008
  cached:                          0               0
  parity:                          0               0
  stripe:                          0               0
  need_gc_gens:                    0               0
  need_discard:                    0               0
  erasure coded:                   0               0
  capacity:             600126783488         1144651

(no label) (device 7):           sdi              rw
                                data         buckets      fragmented
  free:                            0         1108557
  sb:                        3149824               7          520192
  journal:                4294967296            8192
  btree:                    44040192              97         6815744
  user:                  14315700224           27798       258457600
  cached:                          0               0
  parity:                          0               0
  stripe:                          0               0
  need_gc_gens:                    0               0
  need_discard:                    0               0
  erasure coded:                   0               0
  capacity:             600126783488         1144651

(no label) (device 8):           sdj              rw
                                data         buckets      fragmented
  free:                            0         1108555
  sb:                        3149824               7          520192
  journal:                4294967296            8192
  btree:                    46137344             101         6815744
  user:                  14316105728           27796       257003520
  cached:                          0               0
  parity:                          0               0
  stripe:                          0               0
  need_gc_gens:                    0               0
  need_discard:                    0               0
  erasure coded:                   0               0
  capacity:             600126783488         1144651

  
  
bcachefs show-super /dev/sdb
External UUID:                  20c5609b-403d-4e73-b4b9-efbda35fa7af
Internal UUID:                  a70eeb24-f543-48dd-86af-473c08c4246d
Device index:                   0
Label:                          
Version:                        backpointers
Oldest version on disk:         backpointers
Created:                        Thu Jun 23 09:13:50 2022
Sequence number:                33
Superblock size:                6160
Clean:                          0
Devices:                        9
Sections:                       members,replicas_v0,clean,journal_v2,counters
Features:                       new_siphash,new_extent_overwrite,btree_ptr_v2,extents_above_btree_updates,btree_updates_journalled,new_varint,journal_no_flush,alloc_v2,extents_across_btree_nodes
Compat features:                alloc_info,alloc_metadata,extents_above_btree_updates_done,bformat_overflow_done

Options:
  block_size:                   512 B
  btree_node_size:              256 KiB
  errors:                       continue [ro] panic
  metadata_replicas:            1
  data_replicas:                1
  metadata_replicas_required:   1
  data_replicas_required:       1
  encoded_extent_max:           64.0 KiB
  metadata_checksum:            none [crc32c] crc64 xxhash
  data_checksum:                none [crc32c] crc64 xxhash
  compression:                  [none] lz4 gzip zstd
  background_compression:       [none] lz4 gzip zstd
  str_hash:                     crc32c crc64 [siphash]
  metadata_target:              none
  foreground_target:            none
  background_target:            none
  promote_target:               none
  erasure_code:                 0
  inodes_32bit:                 1
  shard_inode_numbers:          1
  inodes_use_key_cache:         1
  gc_reserve_percent:           8
  gc_reserve_bytes:             0 B
  root_reserve_percent:         0
  wide_macs:                    0
  acl:                          1
  usrquota:                     0
  grpquota:                     0
  prjquota:                     0
  journal_flush_delay:          1000
  journal_flush_disabled:       0
  journal_reclaim_delay:        100

members (size 512):
  Device:                       0
    UUID:                       81f88b08-7cb1-4f5b-8a48-c439750399f6
    Size:                       559 GiB
    Bucket size:                512 KiB
    First bucket:               0
    Buckets:                    1144651
    Last mount:                 Thu Jun 23 11:41:27 2022
    State:                      rw
    Label:                      (none)
    Data allowed:               journal,btree,user
    Has data:                   journal,btree,user
    Discard:                    0
    Freespace initialized:      1
  Device:                       1
    UUID:                       bd02b8e4-8739-4bba-ae3f-e12ce62a9b6a
    Size:                       559 GiB
    Bucket size:                512 KiB
    First bucket:               0
    Buckets:                    1144651
    Last mount:                 Thu Jun 23 11:41:27 2022
    State:                      rw
    Label:                      (none)
    Data allowed:               journal,btree,user
    Has data:                   journal,btree,user
    Discard:                    0
    Freespace initialized:      1
  Device:                       2
    UUID:                       25ddac2c-e138-49db-94af-2cfc6d96556b
    Size:                       559 GiB
    Bucket size:                512 KiB
    First bucket:               0
    Buckets:                    1144651
    Last mount:                 Thu Jun 23 11:41:27 2022
    State:                      rw
    Label:                      (none)
    Data allowed:               journal,btree,user
    Has data:                   journal,btree,user
    Discard:                    0
    Freespace initialized:      1
  Device:                       3
    UUID:                       74926f82-7e39-4095-9d2a-24556671706c
    Size:                       559 GiB
    Bucket size:                512 KiB
    First bucket:               0
    Buckets:                    1144651
    Last mount:                 Thu Jun 23 11:41:27 2022
    State:                      rw
    Label:                      (none)
    Data allowed:               journal,btree,user
    Has data:                   journal,btree,user
    Discard:                    0
    Freespace initialized:      1
  Device:                       4
    UUID:                       bda4121f-7d7b-4704-a51b-feaa5abf20ad
    Size:                       559 GiB
    Bucket size:                512 KiB
    First bucket:               0
    Buckets:                    1144651
    Last mount:                 Thu Jun 23 11:41:27 2022
    State:                      rw
    Label:                      (none)
    Data allowed:               journal,btree,user
    Has data:                   journal,btree,user
    Discard:                    0
    Freespace initialized:      1
  Device:                       5
    UUID:                       2ba00a0f-a391-4171-bb53-2c49fb5c704d
    Size:                       559 GiB
    Bucket size:                512 KiB
    First bucket:               0
    Buckets:                    1144651
    Last mount:                 Thu Jun 23 11:41:27 2022
    State:                      rw
    Label:                      (none)
    Data allowed:               journal,btree,user
    Has data:                   journal,btree,user
    Discard:                    0
    Freespace initialized:      1
  Device:                       6
    UUID:                       b070c610-f1c8-4a8a-88e6-b65967b727e5
    Size:                       559 GiB
    Bucket size:                512 KiB
    First bucket:               0
    Buckets:                    1144651
    Last mount:                 Thu Jun 23 11:41:27 2022
    State:                      rw
    Label:                      (none)
    Data allowed:               journal,btree,user
    Has data:                   journal,btree,user
    Discard:                    0
    Freespace initialized:      1
  Device:                       7
    UUID:                       2bc2c24e-01e2-4bc8-a712-14ba87924ad8
    Size:                       559 GiB
    Bucket size:                512 KiB
    First bucket:               0
    Buckets:                    1144651
    Last mount:                 Thu Jun 23 11:41:27 2022
    State:                      rw
    Label:                      (none)
    Data allowed:               journal,btree,user
    Has data:                   journal,btree,user
    Discard:                    0
    Freespace initialized:      1
  Device:                       8
    UUID:                       cc8c9e65-e220-4825-a591-0a9e0bc4e706
    Size:                       559 GiB
    Bucket size:                512 KiB
    First bucket:               0
    Buckets:                    1144651
    Last mount:                 Thu Jun 23 11:41:27 2022
    State:                      rw
    Label:                      (none)
    Data allowed:               journal,btree,user
    Has data:                   journal,btree,user
    Discard:                    0
    Freespace initialized:      1

```
**Kernel bugs**
Compile the kernel with these flags:

I attached the output from `perf trace stream -e bcachefs:*` and `trace-cmd record -e "bcachefs*'` which you can view with `trace-cmd report <FILENAME>`
https://drive.google.com/drive/folders/1l2cz0f5gfGEXlXRgm60Q8xzXjb5jIko9

**Comment by jpsollie** [comment]
it's definitely a locking issue (very funny to debug), but this may help you:
compile the kernel with debug info, and under "hacking options", enable lock debugging:
```
Lock usage statistics
RT Mutex debugging, deadlock detection
Spinlock and rw-lock debugging: basic checks
Mutex debugging: basic checks
RW Semaphore debugging: basic checks
Lock debugging: detect incorrect freeing of live locks
```
after that, recompile your kernel and try again, it may give useful info in dmesg about what the lock issue is all about

**Comment by colttt** [comment]
Hi, 
it's done, in attached you find the dmesg, but I wonder why are there no newer entries about bcachefs locking, because `top` still show bcachefs (like above). Also the link there are new trace-files, maybe its more usefull

[dmesg.log](https://github.com/koverstreet/bcachefs/files/8989215/dmesg.log)

**Comment by colttt** [comment]
Hello,

it still happens :-(
```
uname -a
Linux bcachefs 5.19.0-b2d19c24fdc9-bcachefs #1 SMP PREEMPT_DYNAMIC Tue Aug 16 09:26:57 CEST 2022 x86_64 GNU/Linux
```

in attached the dmesg output and I also updated the google drive link.
[dmesg.log](https://github.com/koverstreet/bcachefs/files/9357541/dmesg.log)

**Comment by jpsollie** [comment]
``
perf_5.10: page allocation failure
``
This is weird ... Looks like bcachefs can't get anny additional memory
@koverstreet : any comments on this? Looks like the user has plenty of RAM(32gb). Any clue why this happens?

**Comment by colttt** [comment]
I hoped that is just happens with multiple devices, but it also happens with a single device:
dmesg output:
```
[  528.704769] bcachefs (sdb): recovering from clean shutdown, journal seq 519
[  528.772450] bcachefs (sdb): going read-write
[  528.795516] bcachefs (sdb): mounted version=backpointers opts=noinodes_use_key_cache
[ 1571.701906] INFO: task kworker/u65:1:963 blocked for more than 120 seconds.
[ 1571.701942]       Tainted: G            E     5.19.0-b2d19c24fdc9-bcachefs #1
[ 1571.701965] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[ 1571.701988] task:kworker/u65:1   state:D stack:    0 pid:  963 ppid:     2 flags:0x00004000
[ 1571.701993] Workqueue: writeback wb_workfn (flush-bcachefs-2)
[ 1571.702002] Call Trace:
[ 1571.702005]  <TASK>
[ 1571.702011]  __schedule+0x370/0xb50
[ 1571.702023]  schedule+0x4e/0xb0
[ 1571.702027]  schedule_timeout+0x17a/0x1c0
[ 1571.702032]  ? lock_release+0x153/0x440
[ 1571.702043]  __down_common+0x13d/0x270
[ 1571.702051]  down+0x43/0x60
[ 1571.702055]  bch2_writepage_do_io+0x20/0x90
[ 1571.702063]  __bch2_writepage+0x382/0x7e0
[ 1571.702070]  ? find_held_lock+0x32/0x90
[ 1571.702073]  ? sched_clock_cpu+0xb/0xc0
[ 1571.702081]  ? __bch2_writepage+0x11d/0x7e0
[ 1571.702089]  write_cache_pages+0x1aa/0x500
[ 1571.702095]  ? bch2_seek_data+0x5e0/0x5e0
[ 1571.702110]  bch2_writepages+0x8e/0xe0
[ 1571.702120]  do_writepages+0xd2/0x1d0
[ 1571.702123]  ? reacquire_held_locks+0xd4/0x1f0
[ 1571.702126]  ? writeback_sb_inodes+0x15c/0x500
[ 1571.702134]  ? find_held_lock+0x32/0x90
[ 1571.702137]  ? sched_clock_cpu+0xb/0xc0
[ 1571.702143]  __writeback_single_inode+0x58/0x480
[ 1571.702150]  writeback_sb_inodes+0x218/0x500
[ 1571.702164]  __writeback_inodes_wb+0x4c/0xe0
[ 1571.702171]  wb_writeback+0x2dd/0x360
[ 1571.702181]  wb_workfn+0x2ec/0x5e0
[ 1571.702191]  process_one_work+0x29d/0x520
[ 1571.702203]  ? rescuer_thread+0x370/0x370
[ 1571.702207]  worker_thread+0x52/0x3a0
[ 1571.702213]  ? rescuer_thread+0x370/0x370
[ 1571.702217]  kthread+0xe7/0x110
[ 1571.702223]  ? kthread_complete_and_exit+0x20/0x20
[ 1571.702228]  ret_from_fork+0x22/0x30
[ 1571.702243]  </TASK>
[ 1571.702252] 
               Showing all locks held in the system:
[ 1571.702257] 4 locks held by kworker/u65:10/195:
[ 1571.702260]  #0: ffff8bc940059d48 ((wq_completion)events_unbound){....}-{0:0}, at: process_one_work+0x206/0x520
[ 1571.702270]  #1: ffffa01803a3be70 ((work_completion)(&rbio->work)){....}-{0:0}, at: process_one_work+0x206/0x520
[ 1571.702279]  #2: ffff8bc953e841a8 (&c->btree_trans_barrier){....}-{0:0}, at: __bch2_trans_init+0xf6/0x290
[ 1571.702290]  #3: ffff8bca839ce890 (b->c.lock){....}-{0:0}, at: __bch2_btree_node_lock+0x477/0x4a0
[ 1571.702302] 4 locks held by kworker/u65:11/200:
[ 1571.702304]  #0: ffff8bc940059d48 ((wq_completion)events_unbound){....}-{0:0}, at: process_one_work+0x206/0x520
[ 1571.702312]  #1: ffffa01803a63e70 ((work_completion)(&rbio->work)){....}-{0:0}, at: process_one_work+0x206/0x520
[ 1571.702320]  #2: ffff8bc953e841a8 (&c->btree_trans_barrier){....}-{0:0}, at: __bch2_trans_init+0xf6/0x290
[ 1571.702328]  #3: ffff8bca839ce890 (b->c.lock){....}-{0:0}, at: __bch2_btree_node_lock+0x477/0x4a0
[ 1571.702337] 1 lock held by khungtaskd/208:
[ 1571.702338]  #0: ffffffff9b2ddac0 (rcu_read_lock){....}-{1:2}, at: debug_show_all_locks+0x15/0xf7
[ 1571.702348] 4 locks held by kworker/10:1/241:
[ 1571.702349]  #0: ffff8bc94837f348 ((wq_completion)bcachefs){....}-{0:0}, at: process_one_work+0x206/0x520
[ 1571.702358]  #1: ffffa018046afe70 ((work_completion)(&cl->work)#4){....}-{0:0}, at: process_one_work+0x206/0x520
[ 1571.702368]  #2: ffff8bc953e841a8 (&c->btree_trans_barrier){....}-{0:0}, at: __bch2_trans_init+0xf6/0x290
[ 1571.702376]  #3: ffff8bca839ce890 (b->c.lock){....}-{0:0}, at: __bch2_btree_node_lock+0x477/0x4a0
[ 1571.702384] 4 locks held by kworker/14:1/244:
[ 1571.702386]  #0: ffff8bc94837f348 ((wq_completion)bcachefs){....}-{0:0}, at: process_one_work+0x206/0x520
[ 1571.702394]  #1: ffffa018046c7e70 ((work_completion)(&cl->work)#4){....}-{0:0}, at: process_one_work+0x206/0x520
[ 1571.702403]  #2: ffff8bc953e841a8 (&c->btree_trans_barrier){....}-{0:0}, at: __bch2_trans_init+0xf6/0x290
[ 1571.702411]  #3: ffff8bca839ce890 (b->c.lock){....}-{0:0}, at: __bch2_btree_node_lock+0x477/0x4a0
[ 1571.702423] 4 locks held by kworker/16:2/739:
[ 1571.702425] 1 lock held by in:imklog/780:
[ 1571.702427]  #0: ffff8bcccb0aeaf8 (&f->f_pos_lock){....}-{3:3}, at: __fdget_pos+0x67/0xb0
[ 1571.702437] 4 locks held by kworker/u65:0/935:
[ 1571.702438]  #0: ffff8bc940059d48 ((wq_completion)events_unbound){....}-{0:0}, at: process_one_work+0x206/0x520
[ 1571.702447]  #1: ffffa01805687e70 ((work_completion)(&rbio->work)){....}-{0:0}, at: process_one_work+0x206/0x520
[ 1571.702455]  #2: ffff8bc953e841a8 (&c->btree_trans_barrier){....}-{0:0}, at: __bch2_trans_init+0xf6/0x290
[ 1571.702463]  #3: ffff8bca839ce890 (b->c.lock){....}-{0:0}, at: __bch2_btree_node_lock+0x477/0x4a0
[ 1571.702471] 3 locks held by kworker/u65:1/963:
[ 1571.702473]  #0: ffff8bc942d80948 ((wq_completion)writeback){....}-{0:0}, at: process_one_work+0x206/0x520
[ 1571.702482]  #1: ffffa01805837e70 ((work_completion)(&(&wb->dwork)->work)){....}-{0:0}, at: process_one_work+0x206/0x520
[ 1571.702490]  #2: ffff8bc9611d80e8 (&type->s_umount_key#49){....}-{3:3}, at: trylock_super+0x16/0x50
[ 1571.702503] 1 lock held by dmesg/1047:
[ 1571.702505]  #0: ffff8bc9bcc780e0 (&user->lock){....}-{3:3}, at: devkmsg_read+0x4b/0x230
[ 1571.702516] 4 locks held by kworker/u65:2/1086:
[ 1571.702517]  #0: ffff8bc940059d48 ((wq_completion)events_unbound){....}-{0:0}, at: process_one_work+0x206/0x520
[ 1571.702526]  #1: ffffa0180688be70 ((work_completion)(&rbio->work)){....}-{0:0}, at: process_one_work+0x206/0x520
[ 1571.702534]  #2: ffff8bc953e841a8 (&c->btree_trans_barrier){....}-{0:0}, at: __bch2_trans_init+0xf6/0x290
[ 1571.702542]  #3: ffff8bca839ce890 (b->c.lock){....}-{0:0}, at: __bch2_btree_node_lock+0x477/0x4a0
[ 1571.702551] 5 locks held by kworker/u65:3/1087:
[ 1571.702553] 10 locks held by kworker/u65:4/1089:
[ 1571.702555] 4 locks held by kworker/u65:5/1090:
[ 1571.702557]  #0: ffff8bc940059d48 ((wq_completion)events_unbound){....}-{0:0}, at: process_one_work+0x206/0x520
[ 1571.702565]  #1: ffffa018068abe70 ((work_completion)(&rbio->work)){....}-{0:0}, at: process_one_work+0x206/0x520
[ 1571.702574]  #2: ffff8bc953e841a8 (&c->btree_trans_barrier){....}-{0:0}, at: __bch2_trans_init+0xf6/0x290
[ 1571.702582]  #3: ffff8bca839ce890 (b->c.lock){....}-{0:0}, at: __bch2_btree_node_lock+0x477/0x4a0
[ 1571.702590] 4 locks held by kworker/u65:6/1103:
[ 1571.702593] 4 locks held by kworker/20:0/1120:
[ 1571.702594]  #0: ffff8bc94837f348 ((wq_completion)bcachefs){....}-{0:0}, at: process_one_work+0x206/0x520
[ 1571.702603]  #1: ffffa01803913e70 ((work_completion)(&cl->work)#4){....}-{0:0}, at: process_one_work+0x206/0x520
[ 1571.702612]  #2: ffff8bc953e841a8 (&c->btree_trans_barrier){....}-{0:0}, at: __bch2_trans_init+0xf6/0x290
[ 1571.702620]  #3: ffff8bca839ce890 (b->c.lock){....}-{0:0}, at: __bch2_btree_node_lock+0x477/0x4a0

[ 1571.702631] =============================================

[ 1813.356875] INFO: task kworker/u65:1:963 blocked for more than 120 seconds.
[ 1813.356919]       Tainted: G            E     5.19.0-b2d19c24fdc9-bcachefs #1
[ 1813.356943] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[ 1813.356973] task:kworker/u65:1   state:D stack:    0 pid:  963 ppid:     2 flags:0x00004000
[ 1813.356982] Workqueue: writeback wb_workfn (flush-bcachefs-2)
[ 1813.356992] Call Trace:
[ 1813.356995]  <TASK>
[ 1813.357001]  __schedule+0x370/0xb50
[ 1813.357014]  schedule+0x4e/0xb0
[ 1813.357020]  schedule_timeout+0x17a/0x1c0
[ 1813.357029]  ? lock_release+0x153/0x440
[ 1813.357039]  __down_common+0x13d/0x270
[ 1813.357048]  down+0x43/0x60
[ 1813.357054]  bch2_writepage_do_io+0x20/0x90
[ 1813.357063]  __bch2_writepage+0x382/0x7e0
[ 1813.357071]  ? find_held_lock+0x32/0x90
[ 1813.357074]  ? sched_clock_cpu+0xb/0xc0
[ 1813.357082]  ? __bch2_writepage+0x11d/0x7e0
[ 1813.357096]  write_cache_pages+0x1aa/0x500
[ 1813.357102]  ? bch2_seek_data+0x5e0/0x5e0
[ 1813.357118]  bch2_writepages+0x8e/0xe0
[ 1813.357135]  do_writepages+0xd2/0x1d0
[ 1813.357138]  ? reacquire_held_locks+0xd4/0x1f0
[ 1813.357141]  ? writeback_sb_inodes+0x15c/0x500
[ 1813.357149]  ? find_held_lock+0x32/0x90
[ 1813.357152]  ? sched_clock_cpu+0xb/0xc0
[ 1813.357159]  __writeback_single_inode+0x58/0x480
[ 1813.357171]  writeback_sb_inodes+0x218/0x500
[ 1813.357186]  __writeback_inodes_wb+0x4c/0xe0
[ 1813.357194]  wb_writeback+0x2dd/0x360
[ 1813.357209]  wb_workfn+0x2ec/0x5e0
[ 1813.357220]  process_one_work+0x29d/0x520
[ 1813.357234]  ? rescuer_thread+0x370/0x370
[ 1813.357240]  worker_thread+0x52/0x3a0
[ 1813.357247]  ? rescuer_thread+0x370/0x370
[ 1813.357251]  kthread+0xe7/0x110
[ 1813.357257]  ? kthread_complete_and_exit+0x20/0x20
[ 1813.357262]  ret_from_fork+0x22/0x30
[ 1813.357283]  </TASK>
[ 1813.357290] 
               Showing all locks held in the system:
[ 1813.357293] 4 locks held by kworker/4:0/39:
[ 1813.357299] 1 lock held by khungtaskd/208:
[ 1813.357301]  #0: ffffffff9b2ddac0 (rcu_read_lock){....}-{1:2}, at: debug_show_all_locks+0x15/0xf7
[ 1813.357318] 4 locks held by kworker/6:1/237:
[ 1813.357321] 5 locks held by kworker/8:1/239:
[ 1813.357323] 4 locks held by kworker/10:1/241:
[ 1813.357326] 4 locks held by kworker/12:1/243:
[ 1813.357327]  #0: ffff8bc94837f348 ((wq_completion)bcachefs){....}-{0:0}, at: process_one_work+0x206/0x520
[ 1813.357337]  #1: ffffa018046bfe70 ((work_completion)(&cl->work)#4){....}-{0:0}, at: process_one_work+0x206/0x520
[ 1813.357353]  #2: ffff8bc953e841a8 (&c->btree_trans_barrier){....}-{0:0}, at: __bch2_trans_init+0xf6/0x290
[ 1813.357364]  #3: ffff8bca839ce890 (b->c.lock){....}-{0:0}, at: __bch2_btree_node_lock+0x477/0x4a0
[ 1813.357376] 1 lock held by in:imklog/780:
[ 1813.357379]  #0: ffff8bcccb0aeaf8 (&f->f_pos_lock){....}-{3:3}, at: __fdget_pos+0x67/0xb0
[ 1813.357394] 3 locks held by kworker/u65:1/963:
[ 1813.357396]  #0: ffff8bc942d80948 ((wq_completion)writeback){....}-{0:0}, at: process_one_work+0x206/0x520
[ 1813.357405]  #1: ffffa01805837e70 ((work_completion)(&(&wb->dwork)->work)){....}-{0:0}, at: process_one_work+0x206/0x520
[ 1813.357415]  #2: ffff8bc9611d80e8 (&type->s_umount_key#49){....}-{3:3}, at: trylock_super+0x16/0x50
[ 1813.357431] 1 lock held by dmesg/1047:
[ 1813.357433]  #0: ffff8bc9bcc780e0 (&user->lock){....}-{3:3}, at: devkmsg_read+0x4b/0x230
[ 1813.357444] 11 locks held by kworker/0:0/1050:
[ 1813.357447] 2 locks held by kworker/15:0/1133:
[ 1813.357449]  #0: ffff8bc940051f48 ((wq_completion)events_freezable_power_){....}-{0:0}, at: process_one_work+0x206/0x520
[ 1813.357464]  #1: ffffa01806a3fe70 ((work_completion)(&(&ev->dwork)->work)){....}-{0:0}, at: process_one_work+0x206/0x520

[ 1813.357475] =============================================
```

**Comment by colttt** [comment]
ok, with the current master
```
Linux bcachefs 5.19.0-9a9c1f3ed8f9-bcachefs #1 SMP PREEMPT_DYNAMIC Fri Aug 19 11:37:08 CEST 2022 x86_64 GNU/Linux
```
it still happens, but now I see a event on top on top perf:
```
top perf
Samples: 66K of event 'cycles', 4000 Hz, Event count (approx.): 10422547911 lost: 0/0 drop: 0/0
Overhead  Shared Object          Symbol
  26.73%  [kernel]               [k] bch2_dev_usage_read
  11.28%  [kernel]               [k] six_optimistic_spin
   6.45%  [kernel]               [k] __lock_acquire.constprop.0
   3.68%  [kernel]               [k] lock_release
```
maybe it's good to know

**Comment by colttt** [comment]
fixed in cd72194e001b
